x-pack/plugins/uptime/server/lib/requests/get_monitor.ts

```diff
@@ -6,18 +6,38 @@
  */
 
 import { SavedObject, SavedObjectsClientContract } from 'kibana/server';
-import { SyntheticsMonitor } from '../../../common/runtime_types';
-import { syntheticsMonitorType } from '../../lib/saved_objects/synthetics_monitor';
+import { EncryptedSavedObjectsClient } from '../../../../encrypted_saved_objects/server';
+import {
+  SyntheticsMonitorWithSecrets,
+  EncryptedSyntheticsMonitor,
+} from '../../../common/runtime_types';
+import {
+  syntheticsMonitor,
+  syntheticsMonitorType,
+} from '../../lib/saved_objects/synthetics_monitor';
 
 export const getSyntheticsMonitor = async ({
   monitorId,
+  encryptedSavedObjectsClient,
   savedObjectsClient,
 }: {
   monitorId: string;
+  encryptedSavedObjectsClient: EncryptedSavedObjectsClient;
   savedObjectsClient: SavedObjectsClientContract;
-}): Promise<SavedObject<SyntheticsMonitor>> => {
+}): Promise<SavedObject<SyntheticsMonitorWithSecrets>> => {
   try {
-    return await savedObjectsClient.get(syntheticsMonitorType, monitorId);
+    const encryptedMonitor = await savedObjectsClient.get<EncryptedSyntheticsMonitor>(
+      syntheticsMonitorType,
+      monitorId
+    );
+
+    return await encryptedSavedObjectsClient.getDecryptedAsInternalUser<SyntheticsMonitorWithSecrets>(
+      syntheticsMonitor.name,
+      monitorId,
+      {
+        namespace: encryptedMonitor.namespaces?.[0],
```

Comment:
i don't understand this part?

Reply:
The encryption must have context of which namespace to look for the object in. Without this, decryption of monitors in non-default namespaces would not work.